components/ephemeral_storage/ephemeral_storage_service.cc

```diff
@@ -236,7 +236,12 @@ void EphemeralStorageService::CleanupFirstPartyStorageArea(
   content::BrowsingDataRemover* remover = context_->GetBrowsingDataRemover();
   content::BrowsingDataRemover::DataType data_to_remove =
       content::BrowsingDataRemover::DATA_TYPE_COOKIES |
-      content::BrowsingDataRemover::DATA_TYPE_DOM_STORAGE;
+      content::BrowsingDataRemover::DATA_TYPE_CACHE |
+      content::BrowsingDataRemover::DATA_TYPE_MEDIA_LICENSES |
```

Comment:
This one might be a bit risky because it's tied to user access to DRM content.

It's definitely local state for a website, but I don't know enough about how that's used in practice and whether or not it's easily recovered / re-created for the user.

Reply:
This is a place where we should err on clearing more vs less, even if its difficult to recover state.

The state for the site should be as-deleted for this site as if the user did "Clear Browsing Data", but for this site only.

@fmarier are you concerned that `content::BrowsingDataRemover::DATA_TYPE_MEDIA_LICENSES` covers more than what "clear browsing data -> all" does? Or concerned that "clear browsing data -> all" would clear too much?

Reply:
The latter: users losing something they can't recover easily.

Which checkbox in `brave://settings/clearBrowserData` clears the `MEDIA_LICENSES`? Cookies and site data?

It's not a huge deal while this is behind a flag and not enabled by default, but beyond that, it could be a problem if this DRM stuff isn't recoverable (e.g. user buys DRMed content, user closes the tab, user loses access to purchased content). But I don't know enough about how DRM stuff works in the wild and so I'm just flagging it as a risk here.

Reply:
@fmarier it looks like its cleared whenever you clear site data, so i expect the "cookies and site data" option, see 
https://source.chromium.org/chromium/chromium/src/+/main:weblayer/browser/profile_impl.cc;l=296?q=content::BrowsingDataRemover::DATA_TYPE_MEDIA_LICENSES,

@goodov , it looks like some other things in that switch statement that should be added here too

Reply:
> "cookies and site data" option, see
> https://source.chromium.org/chromium/chromium/src/+/main:weblayer/browser/profile_impl.cc;l=296?q=content::BrowsingDataRemover::DATA_TYPE_MEDIA_LICENSES,

the link is not exactly right. The correct one should start with `chrome/...`:
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/browsing_data/chrome_browsing_data_remover_constants.h;l=48-51;drc=5e1ffd63a88e3321ac4c56b5fb38a388f781947d

We see that `chrome/` treats `DATA_TYPE_MEDIA_LICENSES` as a part of "Site data".

> @goodov , it looks like some other things in that switch statement that should be added here too

yep, I need to add [`chrome`-declared](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/browsing_data/chrome_browsing_data_remover_constants.h;l=52-56;drc=5e1ffd63a88e3321ac4c56b5fb38a388f781947d) types cleanup via delegate. Now I only cleanup `content`-declared ones.